rasa_sdk/interfaces.py

```diff
@@ -241,9 +241,6 @@ def form_slots_to_validate(self) -> Dict[Text, Any]:
 
         slots_to_validate = {}
 
-        if not self.active_loop:
```

Comment:
Thanks for catching this! I think we need another safe guard then that the slots were actually set by a form execution. We are creating the temporary tracker [here](https://github.com/RasaHQ/rasa/blob/fa6f13a750cd1678ad1ed5b2ba15fa34c63a1646/rasa/core/actions/forms.py#L418) , which is then sent to the action server.

One safeguard idea: If the action before the slots is not an `ActionExecuted(form_name)`, then return an empty dict.

Reply:
I think we don't need the safe guard, because it searches for the past slots that were set before something else happened, if users want to access it, why not?

Reply:
then we should change the name of the function (this would also need to be changed in the docs). Up to @ArjaanBuijk imo

Reply:
up for renaming (we anyhow, get rid of `form` everywhere)

Reply:
> we anyhow, get rid of form everywhere)

Why? It is a form. No need to rename this because of `loop`

Reply:
it can be used everywhere including `loop`, so it is not only for forms. I'd remove a guard and rename it to smth like `last_set_slots`

Reply:
both options are fine to me (safe guard or renaming and removing safeguard) 👍